focus/operations/src/maintenance/mod.rs

```diff
@@ -328,22 +368,76 @@ impl Runner<'_> {
         info!(?time_period, ?path, "running tasks");
         set_default_git_maintenance_config(path)?;
+
+        let maint_result_metric = "maint_result".to_string();
         match self.run_maint(time_period, path) {
             Ok(MaintResult::Success(status)) => {
                 if status.success() {
                     debug!(?time_period, ?path, "completed maintenance",)
                 } else {
                     warn!(?path, exit_status = ?status, "maintenance failed");
+                    let maint_custom_map =
+                        hashmap! { maint_result_metric => maint_exit_status_metric_helper(status) };
+                    self.add_ti_invocation_message(&maint_custom_map);
                 }
             }
-            Ok(MaintResult::LockFailed) => warn!(?path, "failed to acquire lock"),
+            Ok(MaintResult::LockFailed) => {
+                warn!(?path, "failed to acquire lock");
+                let maint_custom_map =
+                    hashmap! { maint_result_metric => "lock_failed".to_string() };
+                self.add_ti_invocation_message(&maint_custom_map);
+            }
             Err(e) => {
-                warn!(?path, ?e, "failed runing git-maintenance");
+                warn!(?path, ?e, "failed running git-maintenance");
+                let maint_custom_map = hashmap! { maint_result_metric => "git_error".to_string() };
+                self.add_ti_invocation_message(&maint_custom_map);
             }
         }
         Ok(())
     }
 
+    fn add_ti_invocation_message(&self, maint_custom_map: &HashMap<String, String>) {
+        let ti_client = self.app.tool_insights_client();
+        ti_client.get_inner().add_invocation_message(
+            SystemTime::now(),
+            None,
+            Some(maint_custom_map),
+        );
+    }
+
+    fn add_maint_ti_invocation_message(
+        &self,
+        git_maint_result: &MaintResult,
+        git_maint_runtime: Duration,
+        sync_maint_result: crate::sync::SyncStatus,
+        sync_maint_runtime: Duration,
+    ) {
+        let mut maint_custom_map = HashMap::new();
+
+        maint_custom_map.insert(
+            "maint_result".to_string(),
+            match git_maint_result {
+                MaintResult::Success(status) => maint_exit_status_metric_helper(*status),
+                MaintResult::LockFailed => "lock_failed".to_string(),
+            },
+        );
+
+        maint_custom_map.insert(
+            "sync_return_status".to_string(),
+            sync_status_metric_helper(sync_maint_result),
+        );
+        maint_custom_map.insert(
+            "git_maint_duration_sec".to_string(),
+            git_maint_runtime.as_secs_f32().to_string(),
+        );
+        maint_custom_map.insert(
+            "sync_maint_duration_sec".to_string(),
+            sync_maint_runtime.as_secs_f32().to_string(),
+        );
+
+        self.add_ti_invocation_message(&maint_custom_map);
```

Comment:
Since we don't have a variable number of items to insert, we can also write it this way. Up to you which is more readable/maintainable.

```suggestion
        self.add_ti_invocation_message(&hashmap! {
            "maint_result".to_string() => match git_maint_result {
                MaintResult::Success(status) => maint_exit_status_metric_helper(*status),
                MaintResult::LockFailed => "lock_failed".to_string(),
            },
            "sync_return_status".to_string() => sync_status_metric_helper(sync_maint_result),
            "git_maint_duration_sec".to_string() => git_maint_runtime.as_secs_f32().to_string(),
            "sync_maint_duration_sec".to_string() => sync_maint_runtime.as_secs_f32().to_string(),
        });
```

Reply:
Fixed.